add qiskit aer guide

kevinsung opened the pull request:
Fixes #291

javabster commented on docs/verify/simulate-with-qiskit-aer.ipynb:
```diff
@@ -0,0 +1,259 @@
+{
```
<p>when you introduce Qiskit Aer can you emphasise that it is a Qiskit Ecosystem project, and link to <a href="https://qiskit.github.io/ecosystem/" target="_blank">https://qiskit.github.io/ecosystem/</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Qiskit/documentation/pull/302/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

kevinsung replied:
Done.

javabster commented on docs/verify/simulate-with-qiskit-aer.ipynb:
```diff
@@ -0,0 +1,259 @@
+{
```
<p>I think at this point we should link to the Qiskit Aer api reference: <a href="https://qiskit.org/ecosystem/aer/stubs/qiskit_aer.primitives.Estimator.html#estimator" target="_blank">https://qiskit.org/ecosystem/aer/stubs/qiskit_aer.primitives.Estimator.html#estimator</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Qiskit/documentation/pull/302/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

kevinsung replied:
Done.

javabster commented on docs/verify/simulate-with-qiskit-aer.ipynb:
```diff
@@ -0,0 +1,259 @@
+{
```
<p>can you extract the code comment and put it as an admonition component outside the code block? It's quite lengthy so I think better to move it out the code block. Also should the TODO be removed?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Qiskit/documentation/pull/302/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

kevinsung replied:
I don't think Jupyter supports admonition components so I used a paragraph quote.

I added the TODO. Do you think it should be removed? If so we need to remember to update the notebook after Qiskit Aer 0.14. If the notebook is tested in CI that is not an issue because the notebook will simply break.

kevinsung replied:
Ok, I opened https://github.com/Qiskit/documentation/issues/319 and I'll remove the TODO.

javabster commented on docs/verify/simulate-with-qiskit-aer.ipynb:
```diff
@@ -0,0 +1,259 @@
+{
```
<p>for documentation we use the second person voice ("you can do this.." not "one can do this...")</p><p> </p><p>Also we shouldn't call this page a "guide" as i think ti will conflict with what they're calling things in the learning section 🙄 I'd recommend saying "example" instead</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Qiskit/documentation/pull/302/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

kevinsung replied:
I don't think "example" is good since it suggests something smaller than the entire page. What about "page"?

kevinsung replied:
I wrote "page" for now.

javabster replied:
<p>can you also change "In practice, you can employ a variety of error mitigation techniques to counter the effects of the noise, but a discussion of these techniques is outside the scope of this guide" to page pls?</p>

kevinsung replied:
That sounded really weird so I changed both to "article." Hope that's okay.

javabster commented:
@kevinsung if you click on the purple "ReviewNB" button you should be able to see which lines my comments apply to :D

javabster commented:
Overall looks great, @abbycross or @beckykd would you mind doing a quick pass at a language review? You always end up catching typos etc that I miss 😅

abbycross commented:
@kevinsung what magic do you do to skip/unskip checks? 😆 I wanted to turn the preview/setup back on so that I can read through the preview.

abbycross commented:
Would you like me to add this to the table of contents for Verify @kevinsung? 
@javabster are you good with the title as it is, or should we change to what you have in the docs box note: "Noisy simulation with Qiskit Aer"?

abbycross commented on docs/verify/simulate-with-qiskit-aer.ipynb:
```diff
@@ -0,0 +1,255 @@
+   "source": [
+    "# Exact and noisy simulation with Qiskit Aer primitives\n",
+    "\n",
+    "In [Exact simulation with Qiskit primitives](https://docs.quantum-computing.ibm.com/verify/simulate-with-qiskit-primitives), we showed how to use the reference primitives included with Qiskit to perform exact simulation of quantum circuits. Currently existing quantum processors suffer from errors, or noise, so the results of an exact simulation do not necessarily reflect the results you woul ...
```
```suggestion
    "[Exact simulation with Qiskit primitives](simulate-with-qiskit-primitives) demonstrates how to use the reference primitives included with Qiskit to perform exact simulation of quantum circuits. Currently existing quantum processors suffer from errors, or noise, so the results of an exact simulation do not necessarily reflect the results you would expect when running circuits on real hardware. While the reference primitives in Qiskit do not support modeling noise, [Qiskit Aer](https://qiskit.org/ecosystem/aer/) includes implementations of the primitives that do support modeling noise. Qiskit Aer is a high-performance quantum circuit simulator that you can use in place of the reference primitives for better performance and more features. It is part of the [Qiskit Ecosystem](https://qiskit.github.io/ecosystem/). In this page, we demonstrate the use of Qiskit Aer primitives for exact and noisy simulation.\n",
```

kevinsung replied:
Done

abbycross commented on docs/verify/simulate-with-qiskit-aer.ipynb:
```diff
@@ -0,0 +1,255 @@
+   "source": [
+    "This circuit contains parameters to represent the rotation angles for the $R_y$ and $R_z$. When simulating this circuit, we will need to specify explicit values for these parameters. In the next cell, we specify some values for these parameters and use the [Estimator](https://qiskit.org/ecosystem/aer/stubs/qiskit_aer.primitives.Estimator.html) primitive from Qiskit Aer to compute the exact e ...
+    "\n",
+    ">Note: Setting `approximation=True` when initializing the Estimator tells Qiskit Aer to approximate the effect of sampling error rather than actually perform sampling. This makes the simulation much more efficient, and also allows us to calculate the exact expectation value, free of sampling error. After Qiskit Aer 0.14, this will be the default behavior, so we won't need to specify this arg ...
```
```suggestion
    "<Admonition type="note">
Setting `approximation=True` when initializing the Estimator tells Qiskit Aer to approximate the effect of sampling error rather than actually perform sampling. This makes the simulation much more efficient, and also allows us to calculate the exact expectation value, free of sampling error. After Qiskit Aer 0.14, this will be the default behavior, so we won't need to specify this argument. </Admonition>"
```

kevinsung replied:
Done.

abbycross commented on docs/verify/simulate-with-qiskit-aer.ipynb:
```diff
@@ -0,0 +1,255 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This circuit contains parameters to represent the rotation angles for the $R_y$ and $R_z$. When simulating this circuit, we will need to specify explicit values for these parameters. In the next cell, we specify some values for these parameters and use the [Estimator](https://qiskit.org/ecosystem/aer/stubs/qiskit_aer.primitives.Estimator.html) primitive from Qiskit Aer to compute the exact e ...
```
```suggestion
    "This circuit contains parameters to represent the rotation angles for $R_y$ and $R_z$. When simulating this circuit, we need to specify explicit values for these parameters. In the next cell, we specify some values for these parameters and use the [Estimator](https://qiskit.org/ecosystem/aer/stubs/qiskit_aer.primitives.Estimator.html) primitive from Qiskit Aer to compute the exact expectation value of the observable $ZZ \\cdots Z$.\n",
```

kevinsung replied:
Done.

abbycross commented on docs/verify/simulate-with-qiskit-aer.ipynb:
```diff
@@ -0,0 +1,255 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, let's initialize a noise model that includes depolarizing error of 2% on every CX gate. In practice, the error arising from the two-qubit gates, which are CX gates here, are the dominant source of error when running a circuit. See [Building Noise Models](https://docs.quantum-computing.ibm.com/verify/building_noise_models) for an overview of constructing noise models in Qiskit Aer.\n",
```
```suggestion
    "Now, let's initialize a noise model that includes depolarizing error of 2% on every CX gate. In practice, the error arising from the two-qubit gates, which are CX gates here, are the dominant source of error when running a circuit. See [Building noise models](https://docs.quantum-computing.ibm.com/verify/building_noise_models) for an overview of constructing noise models in Qiskit Aer.\n",
```

abbycross commented on docs/verify/simulate-with-qiskit-aer.ipynb:
```diff
@@ -0,0 +1,255 @@
+   "source": [
+    "Now, let's initialize a noise model that includes depolarizing error of 2% on every CX gate. In practice, the error arising from the two-qubit gates, which are CX gates here, are the dominant source of error when running a circuit. See [Building Noise Models](https://docs.quantum-computing.ibm.com/verify/building_noise_models) for an overview of constructing noise models in Qiskit Aer.\n",
+    "\n",
+    "In the next cell, we construct an Estimator that incorporates this noise model and use it to compute the expectation value of the observable."
```
```suggestion
    "In the next cell, we construct an Estimator that incorporates this noise model and uses it to compute the expectation value of the observable."
```

kevinsung replied:
Your suggestion changes the meaning of the sentence. It is we who use the Estimator to compute, not the Estimator that uses the noise model to compute.

abbycross replied:
Got it 👍

abbycross commented on docs/verify/simulate-with-qiskit-aer.ipynb:
```diff
@@ -0,0 +1,255 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This value, 65%, gives a rough estimate of the probability that our final state is correct. It is a conservative estimate because it does not take into account the initial state of the simulation. To get a more concrete estimate of how much our final state deviates from the correct state, let's use the [Sampler](https://qiskit.org/ecosystem/aer/stubs/qiskit_aer.primitives.Sampler.html) primi ...
```
```suggestion
    "This value, 65%, gives a rough estimate of the probability that our final state is correct. It is a conservative estimate because it does not take into account the initial state of the simulation. To get a more concrete estimate of how much our final state deviates from the correct state, let's use the [Sampler](https://qiskit.org/ecosystem/aer/stubs/qiskit_aer.primitives.Sampler.html) primitive to estimate the final measurement probability distributions with and without noise, and then compute the fidelity between these distributions. When running the Sampler, we pass `shots=None` to request that the final distribution does not include any random sampling error."
```

kevinsung replied:
I actually prefer my wording here. Is it actually confusing or grammatically incorrect?

abbycross replied:
I agree it's awkward, but IBM Style guide wants us not to use the subjunctive. How about 
"When running the Sampler, pass `shots=None` to exclude any random sampling error from the final distribution."

kevinsung replied:
I've rewritten it as "When running the Sampler, we pass `shots=None` to request a final distribution that does not include random sampling error."

kevinsung commented:
I've addressed all the comments. In updating the TOC, I also went ahead and reordered and renamed some of the other pages according to the plan outlined at #17

abbycross commented:
> > @kevinsung what magic do you do to skip/unskip checks? 😆 I wanted to turn the preview/setup back on so that I can read through the preview.
 I'm asking @Eric-Arellano if he knows how this happens and how to change it

kevinsung commented:
> > > @kevinsung what magic do you do to skip/unskip checks? 😆 I wanted to turn the preview/setup back on so that I can read through the preview.
> > > I'm asking @Eric-Arellano if he knows how this happens and how to change it

Eric said it's because submitted the PR from a fork. In the future, I'll stop using my fork and just use this repository directly. Do you need me to do it now or can I leave this PR as is?

abbycross commented:
> > > > @kevinsung what magic do you do to skip/unskip checks? 😆 I wanted to turn the preview/setup back on so that I can read through the preview.
> > > > I'm asking @Eric-Arellano if he knows how this happens and how to change it
> 
> Eric said it's because submitted the PR from a fork. In the future, I'll stop using my fork and just use this repository directly. Do you need me to do it now or can I leave this PR as is?

No need to change it now -- I can view a preview locally. Thanks!

abbycross reviewed (approved):
👍
